File: examples/src/Slides/Comments/AddParentComments.cs
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using Aspose.slides.Export
using System
using System.Collections.Generic
import aspose.pydrawing as drawing
using System.Linq
using System.text
using System.Threading.Tasks

namespace CSharp.slides.Comments
{
    class AddParentComments
    {
        public static void Run() {

            #ExStart:AddParentComments
            # The path to the output directory.
            outPptxFile = RunExamples.OutPath

            with slides.Presentation() as pres:
            {
                # Add comment
                ICommentAuthor author1 = pres.CommentAuthors.AddAuthor("Author_1", "A.A.")
                IComment comment1 = author1.Comments.AddComment("comment1", pres.slides[0], new PointF(10, 10), DateTime.Now)

                # Add reply for comment1
                ICommentAuthor author2 = pres.CommentAuthors.AddAuthor("Autror_2", "B.B.")
                IComment reply1 = author2.Comments.AddComment("reply 1 for comment 1", pres.slides[0], new PointF(10, 10), DateTime.Now)
                reply1.ParentComment = comment1

                # Add reply for comment1
                IComment reply2 = author2.Comments.AddComment("reply 2 for comment 1", pres.slides[0], new PointF(10, 10), DateTime.Now)
                reply2.ParentComment = comment1

                # Add reply to reply
                IComment subReply = author1.Comments.AddComment("subreply 3 for reply 2", pres.slides[0], new PointF(10, 10), DateTime.Now)
                subReply.ParentComment = reply2

                IComment comment2 = author2.Comments.AddComment("comment 2", pres.slides[0], new PointF(10, 10), DateTime.Now)
                IComment comment3 = author2.Comments.AddComment("comment 3", pres.slides[0], new PointF(10, 10), DateTime.Now)

                IComment reply3 = author1.Comments.AddComment("reply 4 for comment 3", pres.slides[0], new PointF(10, 10), DateTime.Now)
                reply3.ParentComment = comment3

                # Display hierarchy on console
                slide = pres.slides[0]
                comments = slide.GetSlideComments(None)
                for (i = 0 i < comments.Length i++)
                {
                    IComment comment = comments[i]
                    while (comment.ParentComment != None)
                    {
                        Console.Write("\t")
                        comment = comment.ParentComment
                    }

                    Console.Write("{0} : {1}", comments[i].Author.Name, comments[i].text)
                    print()
                }

                pres.save(outPptxFile + "parent_comment.pptx", slides.export.SaveFormat.PPTX)

                # Remove comment1 and all its replies
                comment1.Remove()

                pres.save(outPptxFile + "remove_comment.pptx", slides.export.SaveFormat.PPTX)
            }
            #ExEnd:AddParentComments
        }
    }
}

File: examples/src/Slides/Comments/AddParentComments.py
import aspose.slides as slides
import aspose.pydrawing as drawing
from datetime import date


#ExStart:AddParentComments
# The path to the output directory.
dataDir = "./examples/data/"
outDir = "./examples/out/"

with slides.Presentation() as pres:
    # Add comment
    author1 = pres.comment_authors.add_author("Author_1", "A.A.")
    comment1 = author1.comments.add_comment("comment1", pres.slides[0], drawing.PointF(10, 10), date.today())

    # Add reply for comment1
    author2 = pres.comment_authors.add_author("Autror_2", "B.b.")
    reply1 = author2.comments.add_comment("reply 1 for comment 1", pres.slides[0], drawing.PointF(10, 10), date.today())
    reply1.parent_comment = comment1

    # Add reply for comment1
    reply2 = author2.comments.add_comment("reply 2 for comment 1", pres.slides[0], drawing.PointF(10, 10), date.today())
    reply2.parent_comment = comment1

    # Add reply to reply
    subReply = author1.comments.add_comment("subreply 3 for reply 2", pres.slides[0], drawing.PointF(10, 10), date.today())
    subReply.parent_comment = reply2

    comment2 = author2.comments.add_comment("comment 2", pres.slides[0], drawing.PointF(10, 10), date.today())
    comment3 = author2.comments.add_comment("comment 3", pres.slides[0], drawing.PointF(10, 10), date.today())

    reply3 = author1.comments.add_comment("reply 4 for comment 3", pres.slides[0], drawing.PointF(10, 10), date.today())
    reply3.parent_comment = comment3

    # Display hierarchy on console
    slide = pres.slides[0]
    comments = slide.get_slide_comments(None)
    for i in range(len(comments)):
        comment = comments[i]
        while comment.parent_comment != None:
            print("\t")
            comment = comment.parent_comment

        print("{0} : {1}".format(comments[i].author.name, comments[i].text))
        print()

    pres.save(outDir + "comments_parent_comment_out.pptx", slides.export.SaveFormat.PPTX)

    # Remove comment1 and all its replies
    comment1.remove()

    pres.save(outDir + "comments_remove_comment_out.pptx", slides.export.SaveFormat.PPTX)
#ExEnd:AddParentComments